Purchases/Public/RCPurchases.h

```diff
@@ -283,16 +283,34 @@ NS_SWIFT_NAME(purchaseProduct(_:_:));
 NS_SWIFT_NAME(purchasePackage(_:_:));
 
 /**
- This method will post all purchases associated with the current App Store account to RevenueCat and become associated with the current `appUserID`. If the receipt is being used by an existing user, the current `appUserID` will be aliased together with the `appUserID` of the existing user. Going forward, either `appUserID` will be able to reference the same user.
+ This method will post all purchases associated with the current App Store account to RevenueCat and become associated with the current `appUserID`.
+ If the receipt is being used by an existing user, the current `appUserID` will be aliased together with the `appUserID` of the existing user.
+ Going forward, either `appUserID` will be able to reference the same user.
 
  You shouldn't use this method if you have your own account system. In that case "restoration" is provided by your app passing
  the same `appUserId` used to purchase originally.
 
- @note This may force your users to enter the App Store password so should only be performed on request of the user. Typically with a button in settings or near your purchase UI.
+ @note This may force your users to enter the App Store password so should only be performed on request of the user.
+ Typically with a button in settings or near your purchase UI. Use silentRestoreTransactionsWithCompletionBlock if you need to restore transactions programatically.
  */
 - (void)restoreTransactionsWithCompletionBlock:(nullable RCReceivePurchaserInfoBlock)completion
 NS_SWIFT_NAME(restoreTransactions(_:));
 
+/**
+ This method will post all purchases associated with the current App Store account to RevenueCat and become associated with the current `appUserID`.
+ If the receipt is being used by an existing user, the current `appUserID` will be aliased together with the `appUserID` of the existing user.
+ Going forward, either `appUserID` will be able to reference the same user.
+
+ You shouldn't use this method if you have your own account system. In that case "restoration" is provided by your app passing
+ the same `appUserId` used to purchase originally.
+
+ @note This method will not trigger a login prompt from App Store. However, if the receipt currently on the device does not contain subscriptions,
```

Comment:
If the receipt is nil, it will trigger a login prompt since it will refresh it right?

Reply:
I think that was the previous behavior before we forced to refresh at all times

Reply:
ohh good catch. if that's the case, maybe it's a good idea to modify this a bit more and ensure that we're _never_ prompting a restore on the silent version

Reply:
I made some updates to make sure that the silent version **never** refreshes